Penteract/Assets/Scripts/DialogueManager.cpp

```diff
@@ -0,0 +1,308 @@
+
+	dialoguesArray[4] = Dialogue(9, "", nullptr);
+
+	
+	dialoguesArray[5] = Dialogue(1, "I think I got it...\nYou can power it up this way,\nthen connect this here...\nand that there...", &dialoguesArray[6]);
+	dialoguesArray[6] = Dialogue(1, "Plug it in our core and...\nWHOAH! Oni, try this!", &dialoguesArray[7]);
+	dialoguesArray[7] = Dialogue(2, "I am not sure about this Fang...\nBut OK, I trust you.", &dialoguesArray[8]);
+	dialoguesArray[8] = Dialogue(10, "", nullptr);
+
+	// SWAP DIALOGUE + ONIMARU TUTORIAL
+	dialoguesArray[9] = Dialogue(1, "Oni, you perform better\nin closed spaces.\nTake the lead here!", &dialoguesArray[10]);
+	dialoguesArray[10] = Dialogue(7, "", &dialoguesArray[11]);
+	dialoguesArray[11] = Dialogue(2, "OK Fang.\nWatch how it is done.", &dialoguesArray[12]);
+	dialoguesArray[12] = Dialogue(6, "", nullptr);
+
+}
+
+void DialogueManager::Update() {
+	if (!fangTextComponent) return;
+	if (!onimaruTextComponent) return;
+	if (!tutorialFang) return;
+	if (!tutorialOnimaru) return;
+	if (!tutorialSwap) return;
+	if (!tutorialUpgrades1) return;
+	if (!tutorialUpgrades2) return;
+	if (!tutorialUpgrades3) return;
+
+	if (activeDialogue) {
+		if (runOpenAnimation) ActivateDialogue(activeDialogue);
+
+		if (Input::GetKeyCodeDown(Input::KEY_F) && !(runOpenAnimation || runChangeAnimation || runCloseAnimation) && activeDialogueObject) {
+			if (activeDialogue->nextDialogue) {
+				runChangeAnimation = true;
+				if ((activeDialogue->character < 5 && activeDialogue->nextDialogue->character >= 5) ||
+					(activeDialogue->character >= 5 && activeDialogue->nextDialogue->character < 5)) {
+					runCloseAnimation = true;
+					runSecondaryOpen = true;
+				}
+			}
+			else runCloseAnimation = true;
+		}
+
+		if (runChangeAnimation && !runCloseAnimation) {
+			ActivateNextDialogue(activeDialogue);
+		}
+
+		if (runCloseAnimation) CloseDialogue(activeDialogue);
+	}
+}
+
+void DialogueManager::SetActiveDialogue(Dialogue* dialogue, bool runAnimation)
+{
+	if (activeDialogueObject) {
+		if (activeDialogueObject->IsActive()) {
+			activeDialogueObject->Disable();
+		}
+	}
+
+	if (dialogue) {
+		if (dialogue->character >= 5) {
+			currentStartPosition = tutorialStartPosition;
+			currentEndPosition = tutorialEndPosition;
+		}
+		else {
+			currentStartPosition = dialogueStartPosition;
+			currentEndPosition = dialogueEndPosition;
+		}
+
+		activeDialogue = dialogue;
+		switch (dialogue->character)
+		{
+		case 1:			// FANG
+			activeDialogueObject = fangTextComponent->GetOwner().GetParent();
+			fangTextComponent->SetText(dialogue->text);
+			break;
+		case 2:			// ONIMARU
+			activeDialogueObject = onimaruTextComponent->GetOwner().GetParent();
+			onimaruTextComponent->SetText(dialogue->text);
+			break;
+		case 3:			// DUKE
+			break;
+		case 4:			// ROSAMONDE
+			break;
+		case 5:			// TUTORIAL Fang
+			activeDialogueObject = tutorialFang;
+			break;
+		case 6:			// TUTORIAL Oni
+			activeDialogueObject = tutorialOnimaru;
```

Comment:
Shouldn't the tutorial for Swap come before Onimaru0s tutorial? I am asking this because I have the impresion that the order of the dialogues are in the same order that the cases in the switch are.